### ISSUE TITLE
Rethink object views

### ISSUE THREAD

**Issue by paulrosenzweig** [metabase/metabase#11215]
Related: #5837

Object views appear in the query builder based on attributes of the query. Specifically, we show object views whenever a query contains a filter to a single entity id (after #11209).

Instead, we should think of object views as a distinct way to explore data. Different types of tables would have different representations for their rows.

cc @salsakran @mazameli @kdoh 

# Questions

## Should objects have a distinct route?

For example, a user could view object 7 on some table at `/databases/3/tables/2/entities/7`.

This would most directly represent a different realm for exploring your data. It would be by default separate from dashboards/questions/etc, and we'd have to think through appropriate connection points between those experiences.

Perhaps, if we want to refocus on the data reference, this could be a good thing to add?

It would also create an easy way to deep-link into your Metabase instance.

## Should object views be a visualization type?

This feels less like a new way to _explore_ data and more like a _present_ data. 

This would have the advantage of letting us use the existing settings system to handle customization. It would also cleanly embed into dashboards.

It falls apart a bit if we try to use it as a way to explore data. For example, if you click an entity key on a table in the query builder, you'd be taken to the object view. Does that change the visualization type of the question? Are there any existing cases where interaction changes the visualization type?

## What are the types of objects should we focus on?

Ideally it should be easy to add more, but I think it'd be helpful to have a few types to anchor this discussion. Maybe our sample data has common enough types to use as examples?

## How should people customize object views?

Would people want general control where they rename things? Or, full a dashboard-like WYSIWYG?

This is probably a later step regardless of what general path we take. 

## Do people want embed object views in dashboards?

We currently have the awkward situation where if you're viewing an object view you can save the question. Returning to that question will show the object view as you'd expect. However, the question appears as a table when embedded in dashboards or viewed on a public question page.

## Do people ever want to send object views in a pulse?

Maybe a situation for this would be an account manager that wants updates on a specific account? Even there, I'd expect an object view wouldn't have all the important information and creating a dashboard parameterized by account ID would be better.

**Comment by flamber** [comment]
Just going to reference #4392, since it covers some of the questions you're asking.

**Comment by rthouvenin** [comment]
My 2 cents: Yes, as a user I want to use object views in dashboards, I found this issue while looking for discussions specifically about this. 

The use case that drove me here is a dashboard that shows different metrics / charts about a specific object (e.g. performance metrics of a mailing), and at the top of the dashboard it would be nice to have a summary of the object that looks nicer than a table row. 

Re-ordering and renaming columns would be useful and picking which associations are displayed. It'd nice also if the "table" adjusts the number of key-value pairs to available width instead of only one keys column and one values column.
